### DISCUSSION TITLE
cutPointsWithLoop cuts two loops when 'on' is set to 'cells'

### DISCUSSION THREAD

**Discussion by p-j-smith** [marcomusy/vedo#633] [Q&A] [answered]
Hi,

Here's the example `cutWithPoints1.py` but with `on` set to `cells`:

```py
"""Set a loop of random points on a sphere
to cut a region of the mesh"""
from vedo import settings, Sphere, Points, show

settings.useDepthPeeling = True

s = Sphere().alpha(0.6).lw(0.1)

# pick a few points on the sphere
sc = s.points()
pts = Points([sc[10], sc[15], sc[129], sc[165]], r=12)

#cut loop region identified by the points
scut = s.clone().cutWithPointLoop(pts, invert=False, on='cells')
scut.c('blue',0.7).lw(0).scale(1.03)

show(s, pts, scut, __doc__, axes=1)
```

This cuts two loops in the mesh - the one defined by the set of points, and a second one on the opposite side of the sphere. Do you know if it's possible to not cut the one on the other side of the sphere? Thank you!

**Comment by marcomusy** [comment, marked as answer]
Uhm the two modes work in very different ways.. so it's not obvious..
What you can do is to create manually a box which defines a region that you want to cut and the use `.cutWithMesh()`:
```python
from vedo import *

settings.useDepthPeeling = True

s = Sphere().alpha(0.6).lw(0.1)

sc = s.points()
pts1 = Points([sc[10], sc[15], sc[129], sc[165], sc[10]], r=12).scale(0.8)
pts2 = Points([sc[10], sc[15], sc[129], sc[165], sc[10]], r=12).scale(1.4)
rib = Ribbon(pts1, pts2).clean().cap().computeNormals()

scut = s.clone().cutWithMesh(rib)

ids = rib.insidePoints(s, returnIds=True)
scut2 = s.clone().deleteCellsByPointIndex(ids)

show([[s, pts1,  rib], scut, scut2], N=3, axes=1)
```
![Screenshot from 2022-04-25 18-03-17](https://user-images.githubusercontent.com/32848391/165128344-6f93f724-254e-4101-8676-9188a3b5111a.png)
I'll think if I find a better way of doing it.

**Reply by p-j-smith** [reply]
Thank you, that's very helpful!
